macros/edr/data_monitoring/monitors_query/column_monitoring_query.sql

```diff
@@ -1,18 +1,17 @@
-{% macro column_monitoring_query(monitored_table_relation, timestamp_column, min_bucket_start, column_obj, column_monitors, where_expression=none, time_bucket=none) %}
-
+{% macro column_monitoring_query(monitored_table_relation, min_bucket_start, column_obj, column_monitors, metric_properties) %}
     {% set full_table_name_str = elementary.quote(elementary.relation_to_full_name(monitored_table_relation)) %}
 
     with buckets as (
-        select edr_bucket_start, edr_bucket_end from ({{ elementary.complete_buckets_cte(time_bucket) }}) results
+        select edr_bucket_start, edr_bucket_end from ({{ elementary.complete_buckets_cte(metric_properties['time_bucket']) }}) results
         {% if min_bucket_start -%}
           where edr_bucket_start >= {{ elementary.cast_as_timestamp(min_bucket_start) }}
         {%- endif %}
     ),
-
+    {% set timestamp_column = metric_properties['timestamp_column'] %}
```

Comment:
Feels a bit anti-pattern that for some of the props we directly access the `metric_properties` and for some we don't. Not crucial as the props are not subjective to change, but I'd probably also access directly.

Reply:
I judged if it's used "a lot " or " a little" - timestamp column is very much in use so I thought it'll look better like this, where_expression is used in only one place so I kept access via metric_properties

But I am open to changing either way , so say your opinion

Reply:
Makes sense.